buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts

```diff
@@ -53,19 +51,21 @@ java {
 
 tasks.withType<JavaCompile>().configureEach {
   with(options) {
-    release.set(minJavaVersionSupported.majorVersion.toInt())
+    release.set(otelJava.minJavaVersionSupported.map { it.majorVersion.toInt() })
     compilerArgs.add("-Werror")
   }
 }
 
 // Groovy and Scala compilers don't actually understand --release option
-tasks.withType<GroovyCompile>().configureEach {
-  sourceCompatibility = minJavaVersionSupported.majorVersion
-  targetCompatibility = minJavaVersionSupported.majorVersion
-}
-tasks.withType<ScalaCompile>().configureEach {
-  sourceCompatibility = minJavaVersionSupported.majorVersion
-  targetCompatibility = minJavaVersionSupported.majorVersion
+afterEvaluate {
+  tasks.withType<GroovyCompile>().configureEach {
+    sourceCompatibility = otelJava.minJavaVersionSupported.get().majorVersion
```

Comment:
Just a thought: can you use `map` here, to assign a provider to `sourceCompatibility` and remove `afterEvaluate`?

Reply:
Unfortunately not, `sourceCompatibility` is old and is not a `Provider` (with kotlin you can easily tell the difference between provider and variable, there's no ability to use `=` with a provider). Hopefully someday